src/index.d.ts

```diff
@@ -1,53 +1,169 @@
 // Copyright (c) Jupyter Development Team.
```

Comment:
The input area code was left in a partially complete state, and can be scrapped.  However, I wouldn't be so quick to scrap the code here, as it's mostly a subset of the interface @blink1073 has added here, from our meeting this week.  All that we need is a little bit more code.